pyccel/parser/semantic.py

```diff
@@ -178,6 +178,8 @@ def __init__(self, inputs, **kwargs):
         # used to store the local variables of a code block needed for garbage collecting
         self._allocs = []
 
+        # used to store code split into multiple lines to be reinserted in the CodeBlock
+        self._additional_exprs = []
         # we use it to detect the current method or function
```

Comment:
@EmilyBourne Is this comment line still meaningful to you? I cannot understand it...

Reply:
This still makes sense to me. When we are visiting a line sometimes it needs to be split into multiple parts. E.g:
```
a = 55 + (c if b else d)
```
Needs to be written as:
```
Dummy = c if b else d
a = 55 + Dummy
```
However when we call `_visit_IfTernaryOperator` we need to return a `PyccelAstNode` (and not a `CodeBlock`) so that the `PyccelAdd` knows how to handle the result. So the additional expression `Dummy = c if b else d` goes in `self._additional_exprs` and is reinserted in the `CodeBlock` once the `_visit` command exits to there (see [here](https://github.com/pyccel/pyccel/blob/devel-issue493/pyccel/parser/semantic.py#L1612-L1613))

Reply:
Thanks for the explanation! But I was referring to `# we use it to detect the current method or function` 😜

Reply:
Ah! Hmm, no. Removed lines don't show up on git blame

Reply:
It predates our versioning tags

Reply:
Shall we remove that line then?

Reply:
Apparently the line below the comment was removed on this commit: 2757f9db4a5d76f468b3d766778886a17d74ef1f
Which was about creating the scoping system (24.11.2018). I will remove the comment

Reply:
Fixed 854ffb68